server/handlers/handlers.go

```diff
@@ -372,6 +371,7 @@ func (ah *ActivityHandler) handleCreatedActivity(msg *clientmodels.Message, subs
 	}
 
 	ah.plugin.GetMetrics().ObserveMessage(metrics.ActionCreated, metrics.ActionSourceMSTeams, isDirectMessage)
+	ah.plugin.GetMetrics().ObserveMessageDelay(metrics.ActionCreated, metrics.ActionSourceMSTeams, isDirectMessage, time.Since(msg.CreateAt))
```

Comment:
We're measuring the time between when we received the message (`time.Now()` implicit via `time.Since()`) and the reported time of the message creation from Teams.